generated/nidaqmx/task.py

```diff
@@ -756,6 +782,181 @@ def register_signal_event(self, signal_type, callback_method):
         """
         self._interpreter.register_signal_event(self._handle, signal_type.value, 0, callback_method, None)
 
+    def register_done_event_ex(self, callback_method: Optional[DONE_EVENT_CALLBACK_EX]) -> None:
+        """
+        Registers a callback function to receive an event when a task stops due
+        to an error or when a finite acquisition task or finite generation task
+        completes execution. A Done event does not occur when a task is stopped
+        explicitly, such as by calling DAQmx Stop Task.
+
+        Args:
+            callback_method (function): Specifies the function that you want
+                DAQmx to call when the event occurs. The function you pass in
+                this parameter must have the following prototype:
+
+                >>> def callback(task: nidaqmx.Task, status: int) -> None:
+                >>>     pass
+
+                Upon entry to the callback, the task parameter contains
+                the task on which the event occurred. The status
+                parameter contains the status of the task when the event
+                occurred. If the status value is negative, it indicates an
+                error. If the status value is zero, it indicates no error.
+                If the status value is positive, it indicates a warning. The
+                callbackData parameter contains the value you passed in the
+                callbackData parameter of this function.
+
+                Passing None for this parameter unregisters the event callback
+                function.
+        """
+        def invoke_callback(task_handle: object, status: int, callback_data: object) -> int:
+            callback_method(self, status)
+            return 0
+
+        self._interpreter.register_done_event(self._handle, 0, invoke_callback, None)
+
+    def register_every_n_samples_acquired_into_buffer_event_ex(
```

Comment:
Make sure to update [this ](https://github.com/ni/nidaqmx-python/blob/c5fddf5c9dd6be9b0d02bcd96338963176515cf3/examples/every_n_samples_event.py) example and any others that are registering callbacks to the new versions.